server/src/main/java/com/cloud/template/TemplateManagerImpl.java

```diff
@@ -2219,7 +2222,7 @@ else if (details != null && !details.isEmpty()) {
             _tmpltDao.saveDetails(template);
         }
 
-        _tmpltDao.update(id, template);
+        _tmpltDao.persist(template);
```

Comment:
Here too?

Reply:
Required since it's changed from createForUpdate to findById

Reply:
Persist is usually used for a new record, I'm concerned if this causes any regression or duplicates in DB.

Reply:
There are several instances where findById has been used prior to persist. Tested and no duplicate DB entries were created

Reply:
I was considering getting this towards 4.15RC2 but the additional changes don't give us confidence - only the conditional check was enough to handle the issue.

Reply:
Makes sense, moving the check up